src/hooks/useTokenBalance.ts

```diff
@@ -8,24 +8,28 @@ import useRefresh from './useRefresh'
 import useLastUpdated from './useLastUpdated'
 
 const useTokenBalance = (tokenAddress: string) => {
-  const [balance, setBalance] = useState(BIG_ZERO)
+  const [balanceState, setBalanceState] = useState({ balance: BIG_ZERO, hasFetchedBalance: false })
```

Comment:
This is fine. However, I have found that it is more scalable and flexible if you add a status. The IFO for example uses
```
export type IfoStatus = 'idle' | 'coming_soon' | 'live' | 'finished'
```

Reply:
Agree. That implies that there will be proper handling for `notFetched`, `isFetching`, `fetched` and `failed` statuses. Seems like we've managed so far to reasonably handle most of these states without explicit statuses so I'm not sure how necessary it is, but certainly this is how I'd expect it to be in "perfect" codebase.

Reply:
Yea as, as Cheems said - we don't have an immediate use for these but it's much nicer.

Added!